hw03-custom-junit/src/main/java/org/otus/TestRunner.java

```diff
@@ -0,0 +1,80 @@
+package org.otus;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.otus.ReflectionHelper.callMethod;
+import static org.otus.ReflectionHelper.instantiate;
+
+public class TestRunner {
+    private static final Logger log = LoggerFactory.getLogger(TestRunner.class);
+
+    @SuppressWarnings("java:S1118")
+    public TestRunner() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static void runTests(String className) {
+        try {
+            Class<?> testClass = Class.forName("org.otus." + className);
+
+            List<Method> beforeMethods = new ArrayList<>();
+            List<Method> testMethods = new ArrayList<>();
+            List<Method> afterMethods = new ArrayList<>();
+
+            initialFromAnnotations(testClass, beforeMethods, testMethods, afterMethods);
+
+            int passed = 0;
+            int failed = 0;
+
+            for (Method testMethod : testMethods) {
+                Object testInstance = instantiate(testClass);
+                try {
+                    for (Method beforeMethod : beforeMethods) {
+                        callMethod(testInstance, beforeMethod.getName());
+                    }
+
+                    callMethod(testInstance, testMethod.getName());
+                    passed++;
+                } catch (Exception e) {
+                    log.error(String.format("Test %s failed: %s", testMethod.getName(), e.getCause()));
```

Comment:
```suggestion
                    log.error(String.format("Test %s failed: %s", testMethod.getName(), e.getCause().getCause()));
```
Надо вызвать второй раз `getCause()`, т.к. вы возникшее исключение обернули в `RuntimeException`

Reply:
Исправил